Fixes #14298 - fix popover on facts page

ares opened the pull request:
Fixing the same issue as was fixed in https://github.com/theforeman/foreman/pull/3352 but on facts page.

@tbrisker would you mind reviewing (you're already familiar with this)

ares committed:
Fixes #14298 - fix popover on facts page

tbrisker commented on app/views/fact_values/index.html.erb:
```diff
@@ -7,7 +7,7 @@ end
 <% title_actions documentation_button('3.5.5FactsandtheENC') %>
 
-<table class="<%= table_css_classes('ellipsis')%>">
+<table class="<%= table_css_classes('table-fixed') %>">
```
Do we actually need popovers anywhere here? if so, give the `<th>`s some width so they will not be evenly distributed and give the `ellipsis` class to any `<td>`s that need tooltips.

ares replied:
solved

ares commented:
Thanks for a quick look. I set the ellipsis and fixed width for the name (see screenshot for an example). Value is handled through `trunc_with_tooltip` helper which is in this case more suitable I think.

![facts1](https://cloud.githubusercontent.com/assets/109773/13946274/e74c5202-f013-11e5-822d-9f9aa6ba4691.png)

ares commented on app/views/fact_values/_fact.html.erb:
```diff
@@ -8,14 +8,13 @@
     <% unless fact.compose %>
       <%= link_to trunc_with_tooltip(fact.value, 40),
                   fact_values_path("search" => "facts.#{fact.name} = \"#{fact.value}\""),
-                  :title => _("Show all %{name} facts where they are equal to %{value}") % { :name => fact.name, :value => fact.value },
```
while not entirely related to the issue, this was displaying two conflicting tooltips, there's not much value in this text anyway so I removed it

ohadlevy replied:
IMHO we can expand some space by limiting the actions header size

On Tue, Mar 22, 2016 at 10:55 AM, Marek Hulán notifications@github.com
wrote:

> In app/views/fact_values/_fact.html.erb
> https://github.com/theforeman/foreman/pull/3356#discussion_r56952837:
> 
> > ```
> >  <%= fact_name(fact, parent) %>
> > ```
> > 
> >    </td>
> >    <td class="fact-col">
> >      <% unless fact.compose %>
> >        <%= link_to trunc_with_tooltip(fact.value, 40),
> >                    fact_values_path("search" => "facts.#{fact.name} = \"#{fact.value}\""),
> > -                  :title => _("Show all %{name} facts where they are equal to %{value}") % { :name => fact.name, :value => fact.value },
> 
> while not entirely related to the issue, this was displaying two
> conflicting tooltips, there's not much value in this text anyway so I
> removed it
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/theforeman/foreman/pull/3356/files/ca3e90848038f61684825d79a98e071e7f7a792b#r56952837

ares replied:
yeah, it's not too related to removing this title but I fixed the action th to 1 col

tbrisker replied:
I think it would be actually better here to leave this tooltip and replace the `trunc_with_tooltip` with `fact.value[0,39]` which explains where this link takes us to. If we want to see the complete value we have the `+` button generated in the next line.

ares commented:
test failures seems to be unrelated

tbrisker commented on app/views/fact_values/_fact.html.erb:
```diff
@@ -8,14 +8,13 @@
         <% end %>
```
I would also add `ellipsis` to this column and remove the `trunc_with_tooltip`.

ares replied:
added

tbrisker commented on app/views/fact_values/_fact.html.erb:
```diff
@@ -8,14 +8,13 @@
       </td>
   <% end %>
-  <td>
+  <td class="ellipsis">
```
This will cause double tooltips for facts with very long names.

ares replied:
true, I haven't found any with such long names even with nested facts from chef but I'll remove it

tbrisker commented on app/views/fact_values/index.html.erb:
```diff
@@ -7,16 +7,16 @@ end
-      <th><%= sort :name, :as => _("Name")  %></th>
+      <th class="col-md-3"><%= sort :name, :as => _("Name")  %></th>
       <th><%= sort :value, :as => s_("FactValue|Value") %></th>
       <th><%= _('Reported at') %></th>
```
This can also be limited to `col-md-2` or even `col-md-1` giving us more space for the interesting data.

ares replied:
one was not enough 
![facts2](https://cloud.githubusercontent.com/assets/109773/13985294/3bae2cba-f0fc-11e5-9a20-974d78bd5a19.png)

tbrisker commented on app/views/fact_values/index.html.erb:
```diff
@@ -7,16 +7,16 @@ end
   <thead>
     <tr>
       <% unless params[:host_id] %>
         <th><%= sort :host, :as => _("Host") %></th>
```
This can also be limited to `col-md-2`.

ares commented:
I knew why I should ask you for the review :-) thanks for very good catches. I believe I fixed everything, could you please take another look?

tbrisker commented on app/views/fact_values/_fact.html.erb:
```diff
@@ -1,8 +1,8 @@
+      <td class="ellipsis">
         <% if fact.host -%>
-            <%= link_to(trunc_with_tooltip(fact.host), host_facts_path(:host_id => fact.host), :title => _("Show %{host} facts") % { :host => fact.host.name }) %>
+            <%= link_to(fact.host, host_facts_path(:host_id => fact.host), :title => _("Show %{host} facts") % { :host => fact.host.name }) %>
```
Now that I think about it, this will also get double tootips....

ares replied:
could the ellipsis thing be improved in future so it does not conflict with other tooltips? this seems pretty limiting to me... I'll remove the title for this column...

tbrisker commented on app/views/fact_values/_fact.html.erb:
```diff
@@ -13,11 +13,11 @@
   <td class="fact-col">
     <% unless fact.compose %>
-      <%= link_to trunc_with_tooltip(fact.value, 40),
+      <%= link_to truncate(fact.value, :length => 60),
```
60 seems to be a bit too much:
![screenshot from 2016-03-23 17-24-25](https://cloud.githubusercontent.com/assets/1540531/13990238/4133faf0-f11c-11e5-8525-f1a30415fefa.png)

ares replied:
interesting, my characters seemed to be less wide :-) ok, setting 50 (even though I guess 50 of `@` would overflow as well

ares commented:
@tbrisker I've fixed both comments, would you give it another try please?

tbrisker commented:
@ares 50 seems to be still a bit too long for me, maybe 45 would be better. I agree we should improve the ellipsis js to be smarter (for example, to also know how to correctly ellipsize stuff like nested hostgroups). Also, maybe change the tooltip for the value to not show the value itself but rather just "Show all %{name} facts with the same value"? it makes the + a bit redundant since the complete value is already present in the tooltip.

ares commented:
> 50 seems to be still a bit too long for me, maybe 45 would be better.

changed to 45

> Also, maybe change the tooltip for the value to not show the value itself but rather just "Show all %{name} facts with the same value"? it makes the + a bit redundant since the complete value is already present in the tooltip.

changed

please take another look

tbrisker commented:
Merged as 10db1166231fdaab06c19ed126887d107eaa458b, thanks @ares!
